winch/codegen/src/codegen/context.rs

```diff
@@ -60,58 +61,66 @@ impl<'a> CodeGenContext<'a> {
         self.regalloc.free_gpr(reg);
     }
 
-    /// Loads the stack top value into a register, if it isn't already one;
-    /// spilling if there are no registers available.
-    pub fn pop_to_reg<M: MacroAssembler>(&mut self, masm: &mut M, size: OperandSize) -> Reg {
-        if let Some(reg) = self.stack.pop_reg() {
-            return reg;
-        }
-
-        let dst = self.any_gpr(masm);
-        let val = self.stack.pop().expect("a value at stack top");
-        Self::move_val_to_reg(val, dst, masm, self.frame, size);
-        dst
-    }
-
-    /// Checks if the stack top contains the given register. The register
-    /// gets allocated otherwise, potentially causing a spill.
-    /// Once the requested register is allocated, the value at the top of the stack
-    /// gets loaded into the register.
-    pub fn pop_to_named_reg<M: MacroAssembler>(
+    /// Loads the stack top value into the next available register, if
+    /// it isn't already one; spilling if there are no registers
+    /// available.  Optionally the caller may specify a specific
+    /// destination register.
+    pub fn pop_to_reg<M: MacroAssembler>(
```

Comment:
The work by `pop_to_reg` and `pop_to_named_reg` was very similar, so I realized I could remove some repetition here through this refactor.